typescript/cli/src/validator/preFlightCheck.ts

```diff
@@ -0,0 +1,136 @@
+import { MerkleTreeHook__factory } from '@hyperlane-xyz/core';
+import { HyperlaneCore, S3Validator } from '@hyperlane-xyz/sdk';
+import { Address, eqAddress } from '@hyperlane-xyz/utils';
+
+import { CommandContext } from '../context/types.js';
+import { errorRed, logBlue, logGreen, warnYellow } from '../logger.js';
+
+export const checkValidatorSetup = async (
+  context: CommandContext,
+  chain: string,
+  validators: Set<Address>,
+) => {
+  const { multiProvider, registry } = context;
+
+  const addresses = await registry.getAddresses();
+
+  const core = HyperlaneCore.fromAddressesMap(addresses, multiProvider);
+
+  const validatorAnnounce = core.getContracts(chain).validatorAnnounce;
+  const merkleTreeHook = MerkleTreeHook__factory.connect(
+    addresses[chain].merkleTreeHook,
+    multiProvider.getProvider(chain),
+  );
+
+  let merkleTreeLatestCheckpointIndex: number | undefined;
+  try {
+    const [_, latestCheckpointIndex] = await merkleTreeHook[
+      'latestCheckpoint()'
+    ]();
+
+    merkleTreeLatestCheckpointIndex = latestCheckpointIndex;
+    logBlue(
+      `\nLatest check point index of incremental merkle tree: ${merkleTreeLatestCheckpointIndex}\n`,
+    );
+  } catch (err) {
+    warnYellow(
+      `❗️ Failed to fetch latest checkpoint index of merkleTreeHook on ${chain} \n`,
+    );
+  }
+
+  let announcedValidators;
+  try {
+    announcedValidators = await validatorAnnounce.getAnnouncedValidators();
+  } catch (err) {
+    errorRed(
+      `❌ Failed to fetch announced validators for ${chain}. Exiting.\n`,
+    );
+    process.exit(1);
+  }
+
+  const confirmedValidators = new Set<Address>();
+  const unconfirmedValidators = new Set<Address>();
+
+  for (const validator of validators) {
+    const matches = announcedValidators.filter((address) =>
+      eqAddress(address, validator),
+    );
+    if (matches.length === 0) {
+      unconfirmedValidators.add(validator);
+    } else {
+      confirmedValidators.add(validator);
+    }
+  }
+
+  const errorSet = new Set<string>();
+  errorSet.add('Validator pre flight check failed:');
```

Comment:
nit: found this slightly confusing as I'd expect the errorSet to be empty if there were no errors instead of having size 1 - maybe this could be just a log message on line 131?